core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -434,7 +435,7 @@ public void run() {
                                     } else {
                                         // its a listed node, light connect to it...
                                         logger.trace("connecting to listed node (light) [{}]", listedNode);
-                                        transportService.connectToNodeLight(listedNode);
+                                        transportService.connectToNodeLight(listedNode, pingTimeout, !ignoreClusterName);
```

Comment:
Same thinking as in the previous [case](https://github.com/elastic/elasticsearch/pull/15971#discussion_r60572921) I think.